Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -0,0 +1,364 @@
+instance of `IsScott`.
+
+A class `Scott` is defined in `Topology.OmegaCompletePartialOrder` and made an instance of a
+topological space by defining the open sets to be those which have characteristic functions which
+are monotone and preserve limits of countable chains (`OmegaCompletePartialOrder.Continuous'`).
+A Scott continuous function between `OmegaCompletePartialOrder`s is always
+`OmegaCompletePartialOrder.Continuous'` (`ScottContinuous.continuous'`). The converse is true in
+some special cases, but not in general ([Domain Theory, 2.2.4][abramsky_gabbay_maibaum_1994]).
+
+## References
+
+* [Gierz et al, *A Compendium of Continuous Lattices*][GierzEtAl1980]
+* [Abramsky and Jung, *Domain Theory*][abramsky_gabbay_maibaum_1994]
+
+## Tags
+
+Scott topology, preorder
+-/
+
+variable (α β : Type*)
+
+open Set
+
+section preorder
+
+variable {α β}
+
+variable [Preorder α] [Preorder β]
+
+/--
+A set `u` is said to be inaccessible by directed joins if, when the least upper bound of a directed
+set `d` lies in `u` then `d` has non-empty intersection with `u`.
+-/
+def DirSupInacc (u : Set α) : Prop :=
+  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a : α⦄, IsLUB d a → a ∈ u → (d ∩ u).Nonempty
+
+namespace Topology
+/--
+The Scott-Hausdorff topology is defined as the topological space where a set `u` is open if, when
+the least upper bound of a directed set `d` lies in `u` then there is a tail of `d` which is a
+subset of `u`.
+-/
+def scottHausdorff : TopologicalSpace α where
+  IsOpen u := ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a : α⦄, IsLUB d a →
+    a ∈ u → ∃ b ∈ d, Ici b ∩ d ⊆ u
+  isOpen_univ := fun d ⟨b, hb⟩ _ _ _ _ => ⟨b, hb, (Ici b ∩ d).subset_univ⟩
+  isOpen_inter s t hs ht d hd₁ hd₂ a hd₃ ha := by
+    obtain ⟨b₁, hb₁d, hb₁ds⟩ := hs hd₁ hd₂ hd₃ ha.1
+    obtain ⟨b₂, hb₂d, hb₂dt⟩ := ht hd₁ hd₂ hd₃ ha.2
+    obtain ⟨c, hcd, hc⟩ := hd₂ b₁ hb₁d b₂ hb₂d
+    refine ⟨c, hcd, ?_⟩
+    · calc
+        Ici c ∩ d ⊆ Ici b₁ ∩ Ici b₂ ∩ d := by
+        { apply inter_subset_inter_left d
+          apply subset_inter (Ici_subset_Ici.mpr hc.1) (Ici_subset_Ici.mpr hc.2) }
+        _ = (Ici b₁ ∩ d) ∩ (Ici b₂ ∩ d) := by rw [inter_inter_distrib_right]
+        _ ⊆ s ∩ t := inter_subset_inter hb₁ds hb₂dt
+  isOpen_sUnion := fun s h d hd₁ hd₂ a hd₃ ⟨s₀, hs₀s, has₀⟩ => by
+    obtain ⟨b, hbd, hbds₀⟩ := h s₀ hs₀s hd₁ hd₂ hd₃ has₀
+    exact ⟨b, hbd, Set.subset_sUnion_of_subset s s₀ hbds₀ hs₀s⟩
+
+lemma ScottHausdorff.isOpen_of_isLowerSet {s : Set α} (h : IsLowerSet s) :
+    scottHausdorff.IsOpen s := fun d ⟨b, hb⟩ _ _ hda ha =>
+      ⟨b, hb,Subset.trans (inter_subset_right (Ici b) d)
+      (fun _ hc => h (mem_upperBounds.mp hda.1 _ hc) ha)⟩
+
+/--
+The Scott topology is defined as the join of the topology of upper sets and the Scott Hausdorff
+topology.
+-/
+def Scott : TopologicalSpace α := upperSet α ⊔ scottHausdorff
+
+lemma upperSet_le_Scott : upperSet α ≤ Scott := le_sup_left
```

Comment:
and don't forget to also change the lemma names
```suggestion
lemma upperSet_le_scott : upperSet α ≤ scott := le_sup_left
```

Reply:
Hopefully this is correct now?